### PR TITLE
Implemented native layouts for android

### PR DESCRIPTION
**Pull request by hshristov**
content-view: Added comments on methods that won't be called anymore in android

view.d.ts - removed all padding properties. They were not supposed to be exposed on View. The are implemented only on LayoutBase

view-common:
 - removed paddings
 - fixed getMeasureWidth and getMeasuredHeight to return only size component
 
view.android - CustomLayoutView default native view is now org.nativescript.widgets.ContentView (it was JS NativeViewGroup)

view.ios - added small layout optimization

frame.android - default native view is now org.nativescript.widgets.ContentView (it was JS NativeViewGroup)

image.android - default native view is now org.nativescript.widgets.ImageView (it was JS NativeViewGroup)

All layout classes:
 - changed base class from Layout to LayoutBase (Layout should be used if we want to implement cross-platform layouts in JavaScript)
 - code refactoring
 - splitted into three files
 - on android we use widgets from org.nativescript.widgets jar
 - measure and arrange logic moved into ios files
 - wrap-layout: itemWidth and itemHeight default value changed to 0 (was Number.NaN). Zero is now considered auto instead of Number.NaN

 scroll-view.android - both scrollviews layouts are now implemented in Java. Scroll position is also preserved in all API levels again done in Java
 
style.ts:
 - implemented batch set of margins, paddings, horizontal & vertical alignments
 - changed PropertyMetadataSettings to None for android to skip unnecessary native call to requestLayout (android will request it anyway)

 style.android:
 - added method that translate JS layout properties into native layout properties
 - added LayoutBase styler to allow setting paddings from CSS
 
number-utils - removed isFiniteNumber method. There is a runtime method that do the same - isFinite.

utils - added MEASURED_SIZE_MASK flag into layout module

dependency-observable - improved null check

### CODE REVIEW

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by enchev** [comment]
run ci

**Comment by enchev** [comment]
run ci

**Comment by vchimev** [comment]
run ci

**Comment by hshristov** [comment]
run ci

**Comment by hamorphis** [comment]
:+1:

File: ui/image/image-common.ts [hunk 1]
```diff
@@ -49,43 +47,18 @@ function onSrcPropertyChanged(data: dependencyObservable.PropertyChangeData) {
...
+
+    // None on purpose. for iOS we trigger it manually if needed. Android layout handles it.
+    public static imageSourceProperty = new dependencyObservable.Property(IMAGE_SOURCE, IMAGE,
+        new proxy.PropertyMetadata(undefined, dependencyObservable.PropertyMetadataSettings.None));
+
+    public static isLoadingProperty = new dependencyObservable.Property(ISLOADING, IMAGE,
+        new proxy.PropertyMetadata(false, dependencyObservable.PropertyMetadataSettings.None));
+
+    public static stretchProperty = new dependencyObservable.Property(STRETCH, IMAGE,
+        new proxy.PropertyMetadata(enums.Stretch.aspectFit, dependencyObservable.PropertyMetadataSettings.AffectsLayout));
```
Comment (vakrilov) [inline ui/image/image-common.ts]: AffectsLayout can be None for android

File: ui/layouts/layout-base.ts [hunk 2]
```diff
@@ -116,10 +104,12 @@ export class Layout extends view.CustomLayoutView implements definition.Layout,
         this.style.paddingLeft = value;
     }
 
-    get clipToBounds(): boolean {
-        return this._getValue(Layout.clipToBoundsProperty);
-    }
-    set clipToBounds(value: boolean) {
-        this._setValue(Layout.clipToBoundsProperty, value);
+    protected onClipToBoundsChanged(oldValue: boolean, newValue: boolean) {
```
Comment (vakrilov) [inline ui/layouts/layout-base.ts]: Not implemented for native layouts(grid, stack etc)

File: ui/styling/stylers.android.ts [hunk 3]
```diff
@@ -99,6 +99,109 @@ export class DefaultStyler implements definition.stylers.Styler {
...
+        var width = params.width * utils.layout.getDisplayDensity();
+        var height = params.height * utils.layout.getDisplayDensity();
+        
+        // If width is not specified set it as WRAP_CONTENT
+        if (width < 0) {
+            width = -2;
+        }
+
+        // If height is not specified set it as WRAP_CONTENT
+        if (lp.height < 0) {
```
Comment (vakrilov) [inline ui/styling/stylers.android.ts]: Use `height` here instead of `lp.height`

File: ui/styling/style.ts [hunk 4]
```diff
@@ -23,6 +23,310 @@ var _handlersCache = {};
...
+
+    style._setValue(nativePaddingsProperty, thickness);
+}
+
+function onPaddingChanged(data: observable.PropertyChangeData) {
+    var thickness = parseThickness(data.newValue);
+    var style = <Style>data.object;
+
+    var valueSource = style._getValueSource(paddingProperty);
+    style._setValue(paddingTopProperty, thickness.top, valueSource);
```
Comment (vakrilov) [inline ui/styling/style.ts]: we can call begin/end style updated here.

File: ui/styling/style.ts [hunk 5]
```diff
@@ -23,6 +23,310 @@ var _handlersCache = {};
...
+    style._setValue(paddingBottomProperty, thickness.bottom, valueSource);
+    style._setValue(paddingLeftProperty, thickness.left, valueSource);
+}
+
+function onMarginChanged(data: observable.PropertyChangeData) {
+    var thickness = parseThickness(data.newValue);
+    var style = <Style>data.object;
+
+    var valueSource = style._getValueSource(marginProperty);
+    style._setValue(marginTopProperty, thickness.top, valueSource);
```
Comment (vakrilov) [inline ui/styling/style.ts]: we can call begin/end style updated here.

**Comment by vakrilov** [comment]
:+1:

**Comment by dtopuzov** [comment]
run ci

**Comment by hshristov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Pull request merged**